src/admin_extra/service_admin_extra_migration.erl

```diff
@@ -0,0 +1,788 @@
+%% @private
+%% @doc
+%% Helper for migrate "vcard" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_vcard(Data :: list()) -> Result :: tuple().
+
+migrate_vcard([]) ->
+  {ok, "Completed the migration of the table 'vcard'"};
+
+migrate_vcard([H|T]) ->
+  {Luser, Lserver} = H#vcard.us,
+  Cols = ["username", "server", "vcard"],
+  Vals = [Luser, Lserver, exml:to_binary(H#vcard.vcard)],
+  Q = ["INSERT INTO vcard ", expand_sql_vals(Cols, Vals), ";"],
+  case mongoose_rdbms:sql_query(?HOST, Q) of
+    {error, Reason} ->
+      io:format("The SQL for 'vcard' has error: ~p for values ~p~n", [Reason, Vals]),
+      migrate_vcard(T);
+    _ ->
+      migrate_vcard(T)
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "vcard_search" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_vcard_search(Data :: list()) -> Result :: tuple().
+
+migrate_vcard_search([]) ->
+  {ok, "Completed the migration of the table 'vcard_search'"};
+
+migrate_vcard_search([H|T]) ->
+  {Luser, Lserver} = H#vcard_search.us,
+  Cols = ["username", "lusername", "server", "fn", "lfn", "family", "lfamily", "given",
+          "lgiven", "middle", "lmiddle", "nickname", "lnickname", "bday", "lbday", "ctry",
+          "lctry", "locality", "llocality", "email", "lemail", "orgname", "lorgname", "orgunit",
+          "lorgunit"],
+  Vals = [Luser, H#vcard_search.luser, Lserver, H#vcard_search.fn, H#vcard_search.lfn,
+          H#vcard_search.family, H#vcard_search.lfamily, H#vcard_search.given, H#vcard_search.lgiven,
+          H#vcard_search.middle, H#vcard_search.lmiddle, H#vcard_search.nickname, H#vcard_search.lnickname,
+          H#vcard_search.bday, H#vcard_search.lbday, H#vcard_search.ctry, H#vcard_search.lctry,
+          H#vcard_search.locality, H#vcard_search.llocality, H#vcard_search.email, H#vcard_search.lemail,
+          H#vcard_search.orgname, H#vcard_search.lorgname, H#vcard_search.orgunit, H#vcard_search.lorgunit],
+  Q = ["INSERT INTO vcard_search ", expand_sql_vals(Cols, Vals), ";"],
+  case mongoose_rdbms:sql_query(?HOST, Q) of
+    {error, Reason} ->
+      io:format("The SQL for 'vcard_search' has error: ~p for values ~p~n", [Reason, Vals]),
+      migrate_vcard_search(T);
+    _ ->
+      migrate_vcard_search(T)
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "users" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_users(Data :: list()) -> Result :: tuple().
+
+migrate_users([]) ->
+  {ok, "Completed the migration of the table 'users'"};
+
+migrate_users([H|T]) ->
+  {Luser, _} = H#passwd.us,
+  Cols = ["username", "password"],
+  Vals = [Luser, H#passwd.password],
```

Comment:
This will work for password stored only in plain text. Passwords stored in SCRAM format needs translation. Take a look at module `mongoose_scaram`.